vm-systemd/qubes-network-uplink.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Qubes network uplink (eth0) setup
+Before=network.target
+After=network-pre.target qubes-iptables.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=/usr/lib/qubes/init/network-uplink-setup.sh
+ExecStop=/usr/lib/qubes/init/network-uplink-stop.sh
+
+[Install]
+WantedBy=multi-user.target
```

Comment:
If we use udev to trigger this, the `Install` section would be redundant.

Reply:
But then, having `After=network.target` in other units will be useless.

Reply:
Good point.  I think the best approach is:

- If the VM has an uplink, this should wait for the uplink to be established.
- If there is no uplink, it should return immediately.

Reply:
So, to summarize:
 - start qubes-network-uplink from udev only, drop `[Install]` section (is `Wants=sys-subsystem-net-devices-eth0.device` necessary then?)
 - drop that flag file hack
 - have _another unit_, with `Before=network.target`, that depending on eth0 existence waits for qubes-network-uplink  startup or not

Is that correct?

Reply:
That’s one approach.

Another approach would be to use `ConditionPathExists=/run/qubes-service/uplink` and `After=sys-subsystem-net-devices-eth0.device`.  I have not tested this, so I am not sure if it actually works, however.

Reply:
I'm pretty sure it won't work. `Condition*` are evaluated just before starting, so it won't cancel the effect of `After=`.

Reply:
That’s unfortunate.  Still, your second approach would work and is what I recommend.